src/sql/workbench/parts/objectExplorer/browser/treeSelectionHandler.ts

```diff
@@ -57,12 +60,18 @@ export class TreeSelectionHandler {
 		let specificSelection = selection[0];
 
 		if (this.isMouseEvent(event)) {
-			if (this._lastClicked === specificSelection) {
+			if (this._lastClicked !== undefined) {
 				clearTimeout(this._clickTimer);
-				sendSelectionEvent(event, selection, true);
-				return;
+				let lastSpecificClick = this._lastClicked[0];
+
+				if (lastSpecificClick === specificSelection) {
+					sendSelectionEvent(event, selection, true);
+					return;
+				} else {
+					sendSelectionEvent(event, this._lastClicked, false);
+				}
 			}
-			this._lastClicked = specificSelection;
+			this._lastClicked = selection;
 		}
 
 		this._clickTimer = setTimeout(() => {
```

Comment:
didn't we need to set last clicked to undefined here?

Reply:
Nah this one needs to keep track of what was just clicked for double click purposes.